File: finbyzreach/api/customer.py
from finbyzreach.utils.research import research_company
import frappe

@frappe.whitelist(methods=["POST"])
def research_customer(doc=None, name=None):
    if doc:
        doc = frappe.parse_json(doc)
        name = doc.get("name")
        if doc.get("company_details"):
            return
    if not name:
        return
    return research_company("Customer", name, enforce_permissions=True)

File: finbyzreach/public/js/customer.js
frappe.ui.form.on("Customer", {
   refresh(frm){
        frm.add_custom_button("Research", function() {
            frappe.call({
                method: "finbyzreach.api.customer.research_customer",
                freeze: true,
			    freeze_message: __("Gethering Information..."),
                args: {
                    name: cur_frm.doc.name
                },
                callback: function(r) {
                    if (r.message) {
                        frm.refresh_fields()
                        frappe.msgprint("Research completed");
                    }
                }
            });
        })
   }, 
});